Add embedding models configurable, from both Xenova and TEI (#646)
huggingface/chat-ui: README.md

1. Generating an appropriate search query from the user prompt.
2. Performing web search and extracting content from webpages.
3. Creating embeddings from texts using a text embedding model.
4. From these embeddings, find the ones that are closest to the user query using a vector similarity search. Specifically, we use `inner product` distance.
5. Get the corresponding texts to those closest embeddings and perform [Retrieval-Augmented Generation](https://huggingface.co/papers/2005.11401) (i.e. expand user prompt by adding those texts so that an LLM can use this information).

> ### Text Embedding Models

Reviewer: ```suggestion
## Text Embedding Models
```

it can be heading 2 and you can add this section to the toctree list [here](https://github.com/huggingface/chat-ui/blob/feature/assistants/README.md#L22)

> By default (for backward compatibility) when not defining TEXT_EMBEDDING_MODELS environment variable It will use [transformers.js](https://huggingface.co/docs/transformers.js), specifically, [Xenova/gte-small](https://huggingface.co/Xenova/gte-small) model.

Reviewer: ```suggestion
By default (for backward compatibility), when `TEXT_EMBEDDING_MODELS` environment variable is not defined, [transformers.js](https://huggingface.co/docs/transformers.js) embedding models will be used for embedding tasks, specifically, [Xenova/gte-small](https://huggingface.co/Xenova/gte-small) model.
```

> You can customize the embedding model by setting TEXT_EMBEDDING_MODELS in your `.env.local`, for example

Reviewer: ```suggestion
You can customize the embedding model by setting `TEXT_EMBEDDING_MODELS` in your `.env.local` file. For example:
```

```env
TEXT_EMBEDDING_MODELS = `[
  {
    "name": "Xenova/gte-small",
    "displayName": "Xenova/gte-small",
    "description": "locally running embedding",
    "maxSequenceLength": 512,
    "endpoints": [
      {"type": "xenova"}
    ]
  },
  {
    "name": "intfloat/e5-base-v2",
    "displayName": "intfloat/e5-base-v2",
    "description": "hosted embedding model",
    "maxSequenceLength": 768,
    "preQuery": "query: ", # See https://huggingface.co/intfloat/e5-base-v2#faq
    "prePassage": "passage: ", # See https://huggingface.co/intfloat/e5-base-v2#faq
    "endpoints": [
      {"type": "tei", "url": "http://127.0.0.1:8080/"}
    ]
  }
]`
```

> The required fields are `name`, `maxSequenceLength` and `endpoints`.
> It supports [transformers.js](https://huggingface.co/docs/transformers.js) and [TEI](https://github.com/huggingface/text-embeddings-inference), transformers.js model run locally, and TEI models run in a different environment. each `endpoints` provided supports a `weight` parameter which will be used to determine the probability of requesting a particular endpoint.

Reviewer: ```suggestion
The required fields are `name`, `maxSequenceLength` and `endpoints`.
Supported text embedding backends are: [`transformers.js`](https://huggingface.co/docs/transformers.js) and [`TEI`](https://github.com/huggingface/text-embeddings-inference). `transformers.js` models run locally as part of `chat-ui`, whereas `TEI` models run in a different environment & accessed through an API endpoint.
```

> When defining more than one embedding model, the first will be used by default, and the others will only be used on LLM's which configured `embeddingModelName` to the name of the model.

Reviewer: ```suggestion
When more than one embedding models are supplied in `.env.local` file, the first will be used by default, and the others will only be used on LLM's which configured `embeddingModelName` to the name of the model.
```

If you're using a self-signed certificate, e.g. for testing or development purposes, you can set the `REJECT_UNAUTHORIZED` parameter to `false` in your `.env.local`. This will disable certificate validation, and allow Chat UI to connect to your custom endpoint.

#### Specific Embedding Model

A model can use any of the embedding models defined in `.env.local`, (currently used when web searching),
by default it will use the first embedding model, but it can be changed with the field `embeddingModel`:

```env
TEXT_EMBEDDING_MODELS = `[
  {
    "name": "Xenova/gte-small",
    "maxSequenceLength": 512,
    "endpoints": [

>       {"type": "xenova"}

Reviewer: ```suggestion
      {"type": "transformersjs"}
```

    ]
  },

>   {

Reviewer: could you add [dummy authorization field ](https://github.com/mikelfried/chat-ui/blob/embedding_models/src/lib/server/embeddingEndpoints/tei/embeddingEndpoints.ts#L51-L52) example here?
